### ISSUE TITLE
Follow up of #13038 - swiping up ad notification doesn't trigger the dismiss event type

### ISSUE THREAD

**Issue by GeetaSarvadnya** [brave/brave-browser#16361]
## Description 

Follow up of #13038 - swiping up ad notification doesn't trigger the dismiss event type
## Steps to reproduce 

Followed the test plan from https://github.com/brave/brave-core/pull/7293

   1. Clean profile 1.26.56
   2. Enable rewards
   3. View a custom AD
   4. Swipe up the AD notification to trigger dismiss an event
   5. Dismiss event type is not recorded in the logs
   
## Actual result 

Swiping up ad notification doesn't trigger the dismiss event type
## Expected result

Dismiss event type should be recorded in the logs

## Issue reproduces how often 
Easy

## Version/Channel Information:

- Can you reproduce this issue with the current Play Store version? No
- Can you reproduce this issue with the current Play Store Beta version? Yes
- Can you reproduce this issue with the current Play Store Nightly version? Not sure

## Device details
   - Install type (ARM, x86): ARM
   - Device type (Phone, Tablet, Phablet): Samsung Galaxy Tablet
   - Android version: 10
   
## Brave version

1.26.56
### Website problems only
- Does the issue resolve itself when disabling Brave Shields? NA
- Does the issue resolve itself when disabling Brave Rewards? NA
- Is the issue reproducible on the latest version of Chrome? NA

### Additional information

cc: @brave/legacy_qa @tmancey

**Comment by srirambv** [comment]
Dismiss event is causing the ad to timeout as per logs 
```
2021-06-14 10:32:03.353 31344-31344/? V/chromium: [VERBOSE1:ads_service_impl.cc(1819)] Timeout ad notification with uuid 4ed68333-3e25-40f4-9aa5-a91696bead0e in 30 seconds
2021-06-14 10:32:03.354 31344-31344/? V/chromium: [VERBOSE1:ad_notification_serving.cc(105)] Ad notification delivered:
      uuid: 4ed68333-3e25-40f4-9aa5-a91696bead0e
      creativeInstanceId: 4cda3ae7-1f70-4bca-b361-ca01d6a8f1ed
      creativeSetId: 5f8c4787-8431-49a5-9d09-bba40f8ad5f4
      campaignId: d1e25388-b362-4f30-bd4b-36b8205e59e5
      advertiserId: a34de75d-5654-4331-8713-8f4d4e2b6bf7
      segment: untargeted
      title: Vivamus sagittis lacus vel aug
      body: Duis mollis, est non commodo luctus, nisi erat porttitor lig
      targetUrl: https://www.apple.com
2021-06-14 10:32:03.387 31344-31344/? V/chromium: [VERBOSE3:ad_notification_event_viewed.cc(24)] Viewed ad notification with uuid 4ed68333-3e25-40f4-9aa5-a91696bead0e and creative instance id 4cda3ae7-1f70-4bca-b361-ca01d6a8f1ed

2021-06-14 10:32:05.747 31344-31344/? V/chromium: [VERBOSE1:confirmations.cc(248)] Successfully redeemed unblinded token with confirmation id 0111fb1f-b407-4c96-bace-ab4b4f58098b, creative instance id 4cda3ae7-1f70-4bca-b361-ca01d6a8f1ed and view
2021-06-14 10:32:05.751 31344-31344/? V/chromium: [VERBOSE1:confirmations.cc(271)] Added 1 unblinded payment token with an estimated redemption value of 0.01 BAT, you now have 1 unblinded payment tokens
2021-06-14 10:32:05.753 31344-31344/? V/chromium: [VERBOSE1:refill_unblinded_tokens.cc(66)] No need to refill unblinded tokens as we already have 49 unblinded tokens which is above the minimum threshold of 20
2021-06-14 10:32:06.321 31344-31344/? V/chromium: [VERBOSE3:ad_notification_event_timed_out.cc(22)] Timed out ad notification with uuid 4ed68333-3e25-40f4-9aa5-a91696bead0e and creative instance id 4cda3ae7-1f70-4bca-b361-ca01d6a8f1ed

```

**Comment by stephendonner** [comment]
**Verified `FIXED`** using `Brave 1.26.63, Chromium 91.0.4472.106` on Google Pixel, Android 9, ARM64

**Steps:**

1. Clean profile
2. Launched `1.26.63`
3. Passed in `--variations-server-url=https://no-thanks.invalid --enable-features="AdNotifications<CommandLineStudy" --force-fieldtrials="*CommandLineStudy/Enabled" --force-fieldtrial-params="CommandLineStudy.Enabled:should_show_custom_notifications/true"` via the `Command Line String` in Debug settings
4. relaunched Brave
5. Enabled Rewards
6. Viewed a custom ad
7. Swiped up on the ad notification to dismiss

Confirmed I got the custom ad:

```
2021-06-16 17:59:53.190 27775-27775/? V/chromium: [VERBOSE1:ad_notification_serving.cc(105)] Ad notification delivered:
      uuid: 26bdab01-202e-446b-a1e9-b136d027083d
      creativeInstanceId: 6f5decfc-57cb-4a42-8523-e372c7d4adff
      creativeSetId: 57388e7d-e0fd-4f30-a76a-3d7fb2e6d8a1
      campaignId: ccdaf616-b8a8-481a-aa8b-caf5fb131353
      advertiserId: aa1d1cdb-ec45-42d4-86d0-ea353e1babd4
      segment: untargeted
      title: A quick demo of verifiable conversions
      body: Hello Brave!
      targetUrl: https://saberistic.com
```
Confirmed it logged `view`s:

```
2021-06-16 17:59:53.205 27775-27775/? V/chromium: [VERBOSE3:ad_notification_event_viewed.cc(24)] Viewed ad notification with uuid 26bdab01-202e-446b-a1e9-b136d027083d and creative instance id 6f5decfc-57cb-4a42-8523-e372c7d4adff
2021-06-16 17:59:53.205 27775-27775/? V/chromium: [VERBOSE1:confirmations.cc(89)] Confirming view ad for creative instance id 6f5decfc-57cb-4a42-8523-e372c7d4adff
2021-06-16 17:59:53.205 27775-27775/? V/chromium: [VERBOSE1:redeem_unblinded_token.cc(50)] Redeem unblinded token
```

```
2021-06-16 17:59:53.225 27775-27775/? V/chromium: [VERBOSE6:ad_notification_event_viewed.cc(32)] Successfully logged ad notification viewed event
```

Finally, confirmed I got the `dismiss` event logged just fine:

```
73d15f-0329-468f-a791-77e7577e302c","createdAt":"2021-06-17T01:00:02.831Z","type":"dismiss","modifiedAt":"2021-06-17T01:00:02.839Z","creativeInstanceId":"6f5decfc-57cb-4a42-8523-e372c7d4adff","paymentToken":{"publicKey":"lLO5tErGoTK0askrALab6pKGAnBHqELSyw/evqZRwH8=","batchProof":"yYKp2NeuzipR4teBDbgATyB1yv2MgDfp8LMd1S8L7giiM7mwaBXR5NR6PMiKyE0iE+cDjrqWvKrv0q8ljSJoDQ==","signedTokens":["XAPw2VYHxK3o/MuElTvOB4MGxR7/ZqyfPnKlyXS+Dlg="]}}
2021-06-16 18:00:02.928 27775-27775/? V/chromium: [VERBOSE1:confirmations.cc(248)] Successfully redeemed unblinded token with confirmation id 8273d15f-0329-468f-a791-77e7577e302c, creative instance id 6f5decfc-57cb-4a42-8523-e372c7d4adff and dismiss
2021-06-16 18:00:02.932 27775-27775/? V/chromium: [VERBOSE1:confirmations.cc(271)] Added 1 unblinded payment token with an estimated redemption value of 0 BAT, you now have 7 unblinded payment tokens
```
